### PR TITLE
Mat 4209 validate value set

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-4209](https://jira.cms.gov/browse/MAT-4209)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [x] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Review by RohitKandimalla — DISMISSED** [review]

File: src/components/measureEditor/MeasureEditor.tsx [hunk 1]
```diff
@@ -78,18 +80,152 @@ const MeasureEditor = () => {
...
+  const [valuesetMsg, setValuesetMsg] = useState(null);
+
+  const getValueSetErrors = async (
+    valuesetsArray: ElmValueSet[],
+    tgtValue: string
+  ): Promise<ElmTranslationError[]> => {
+    const valuesetsErrorArray: ElmTranslationError[] = [];
+    if (valuesetsArray && tgtValue) {
+      const results = await Promise.allSettled(
+        valuesetsArray.map(async (valueSet, i) => {
```
Comment (adongare) [inline src/components/measureEditor/MeasureEditor.tsx:93]: Could this be done in batch instead of calling service for each VS?

Comment (sb-cecilialiu) [reply src/components/measureEditor/MeasureEditor.tsx:93]: @adongare It can, the back end service needs to be changed first. Maybe at a later time?

File: src/components/measureEditor/MeasureEditor.tsx [hunk 2]
```diff
@@ -78,18 +80,152 @@ const MeasureEditor = () => {
...
+    }
+    return tgtValue;
+  };
+
+  const getOid = (valueSet: ElmValueSet): string => {
+    const id = valueSet.id.split("").reverse().join("");
+    const index = id.indexOf("/");
+    const oid = id.substr(0, index).split("").reverse().join("");
+
+    return oid;
```
Comment (adongare) [inline src/components/measureEditor/MeasureEditor.tsx]: I think this can be simplified. there is a regex for oid: https://hl7.org/fhir/datatypes.html#oid
e.g `valueSet.id.match(/[0-2](\.(0|[1-9][0-9]*))+/)[0]` have to check for match(...) results before getting 0th match.

File: src/components/measureEditor/MeasureEditor.tsx (hunk 1 above)
Comment (adongare) [reply src/components/measureEditor/MeasureEditor.tsx:93]: yeah, that works, Maybe create a new tech debt story and throw it in the backlog.

File: src/components/measureEditor/MeasureEditor.tsx [hunk 3]
```diff
@@ -78,18 +80,152 @@ const MeasureEditor = () => {
...
+    return window.localStorage.getItem("TGT");
+  };
+  const getTgtValue = (tgt: any): any => {
+    let tgtValue = null;
+    if (tgt) {
+      let tgtObjFromLocalStorage = JSON.parse(tgt);
+      for (const [key, value] of Object.entries(tgtObjFromLocalStorage)) {
+        if (key === "TGT") {
+          tgtValue = value.toString();
+        }
```
Comment (adongare) [inline src/components/measureEditor/MeasureEditor.tsx]: Probably loop is not required? `tgtObjFromLocalStorage.TGT`

File: src/components/measureEditor/MeasureEditor.tsx (hunk 2 above)
Comment (sb-cecilialiu) [reply src/components/measureEditor/MeasureEditor.tsx]: @adongare That's great info! Thanks a lot! I pushed in the change.

File: src/components/measureEditor/MeasureEditor.tsx (hunk 3 above)
Comment (sb-cecilialiu) [reply src/components/measureEditor/MeasureEditor.tsx]: @adongare This is done too.

File: src/components/measureEditor/MeasureEditor.tsx (hunk 1 above)
Comment (sb-cecilialiu) [reply src/components/measureEditor/MeasureEditor.tsx:93]: @adongare I talked to Joe and our team, there will be a new ticket for it.

**Review by adongare — APPROVED** [review]
